Tests/PackageLoadingTests/JSONSerializationTests.swift

```diff
@@ -54,7 +54,20 @@ class JSONSerializationTests: XCTestCase {
         let t1 = Target(name: "One")
         let t2 = Target(name: "Two", dependencies: [.Target(name: "One")])
         let package = Package(name: "Targets", targets: [t1, t2])
-        assertEqual(package: package, expected: "{\"dependencies\": [], \"exclude\": [], \"name\": \"Targets\", \"targets\": [{\"dependencies\": [], \"name\": \"One\"}, {\"dependencies\": [\"One\"], \"name\": \"Two\"}]}")
+        assertEqual(package: package, expected: "{\"dependencies\": [], \"exclude\": [], \"name\": \"Targets\", \"targets\": [{\"dependencies\": [], \"name\": \"One\"}, {\"dependencies\": [{\"name\": \"One\", \"type\": \"target\"}], \"name\": \"Two\"}]}")
+
+        // Test target dependencies.
+        var dependency: Target.Dependency = .Target(name: "foo")
+        XCTAssertEqual(dependency.toJSON().toString(), "{\"name\": \"foo\", \"type\": \"target\"}")
+
+        dependency = .ByName(name: "foo")
+        XCTAssertEqual(dependency.toJSON().toString(), "{\"name\": \"foo\", \"type\": \"byname\"}")
+
+        dependency = .Product(name: "foo", package: nil)
```

Comment:
Hmm actually its not possible have default params in enum with associated data, so clients will have these options:

```swift
Target(
    name: "Foo",
    dependencies: [
        .Product(name: "Bar", package: nil),
        .Product(name: "Bar", package: "Pkg"),
        "Pkg",
    ]
)
```